src/libraries/System.Data.OleDb/src/OleDbDataReader.cs

```diff
@@ -1220,7 +1220,7 @@ private static IntPtr AddRecordsAffected(IntPtr recordsAffected, IntPtr affected
                 {
```

Comment:
This whole method should be simplified to:
```
        private static nint AddRecordsAffected(nint recordsAffected, nint affected)
        {
            if (0 <= affected)
            {
                if (0 <= recordsAffected)
                {
                    return recordsAffected + affected;
                }
                return affected;
            }
            return recordsAffected;
        }
```

Reply:
All places that use `IsRunningOnX86` in this file can use same simplification. `IsRunningOnX86` is used to switch between int and long. nint will handle both paths.